src/uu/more/src/more.rs

```diff
@@ -239,23 +239,29 @@ fn more(buff: &str, mut stdout: &mut Stdout, next_file: Option<&str>, silent: bo
                     code: KeyCode::Char(' '),
                     modifiers: KeyModifiers::NONE,
                 }) => {
-                    pager.page_down();
+                    if pager.should_close() {
+                        return;
+                    } else {
+                        pager.page_down();
+                    }
```

Comment:
I think the ordering for this should be reversed: First page down, then draw, then return if we should close.

Reply:
If the intention was to differ in behavior from the actual `more`, then I think this is fine. I initially assumed you were trying to match `more`s behavior. (also applies to my other comments)

Reply:
This should fix this https://github.com/uutils/coreutils/pull/2407/commits/9ed5091be6bf4f0f9c13ec85a1b1ea9f72c4eccb